pkg/supportbundle/spec.go

```diff
@@ -144,7 +145,13 @@ func CreateRenderedSpec(app apptypes.AppType, sequence int64, kotsKinds *kotsuti
 	renderedSpec = b.Bytes()
 
 	secretName := GetSpecSecretName(app.GetSlug())
-	existingSecret, err := clientset.CoreV1().Secrets(util.PodNamespace).Get(context.TODO(), secretName, metav1.GetOptions{})
+	// to add the troubleshoot labels to existing support-bundle spec secrets that were created in older versions of kots
+	_, err = k8sutil.AddLabelsToSecret(clientset, util.PodNamespace, secretName, kotstypes.GetTroubleshootLabels())
+	if err != nil {
+		errors.Wrap(err, "failed to add troubleshoot labels to existing spec secret")
+	}
+
+	existingSecret, err := GetSpecSecretsMatchingLabel(clientset, labels.SelectorFromSet(kotstypes.GetTroubleshootLabels()).String(), util.PodNamespace, SpecDataKey)
```

Comment:
wouldn't the troubleshoot labels match multiple secrets at some point? KOTS creates a secret for each application version. the `GetSpecSecretsMatchingLabel` might return a different secret than the one that matches `secretName`. if the secret returned is different, then it could use the wrong spec.

Reply:
By design there would also be other secrets with labels that we _do_ want collected. kURL and kURL add-ons for example will install support bundles to collect information on those components.

I think if KOTS maintains multiple versions of a spec we should be labeling only the on that's considered 'active' and not 'historic' specs because we do want to collect those specs provided by other projects.

Reply:
> KOTS creates a secret for each application version

I'm curious what do you mean by application version here? Are you saying each separate app install or each sequence of an app release?

Reply:
@diamonwiggins each sequence of an app release. e.g. for each version in the "Version history" page

Reply:
actually, looks like my info is outdated. there will be a secret per app, not app version. still a similar issue though. KOTS should be using specific labels that scope the spec to a single app

Reply:
why are we querying the support bundle spec that kots uses by the new troubleshoot label?  we previously queried by name since it is deterministic.  adding the new label to the secret makes sense so that troubleshoot itself can find it outside of kots, but I'm not sure I'm understanding why we need to query for it that way here.

Reply:
I see what you mean @cbodonnell. So basically this PR should only have the logic of adding the troubleshoot label to the secret.

Gotcha